modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/scanner/Ingestor.java

```diff
@@ -239,6 +249,31 @@ public RetriableIngestJob call() {
                   mediaPackage = mediaPackages.get(0);
                   var id = mediaPackage.getIdentifier().toString();
                   var eventConfiguration = schedulerService.getCaptureAgentConfiguration(id);
+
+                  // Check if the scheduled event already has a recording associated with it
+                  // If so, ingest the file as a new event
+                  try {
+                    Recording recordingState = schedulerService.getRecordingState(id);
+                    if (recordingState.getState().equals(UPLOAD_FINISHED)) {
+                      mediaPackage = (MediaPackage) mediaPackage.clone();
+                      Id newId = IdImpl.fromUUID();
+                      mediaPackage.setIdentifier(newId);
+
+                      // Update dublincore title
+                      try {
+                        String newTitle = mediaPackage.getTitle() + " - " + newId;
```

Comment:
Some insider knowledge: Never try getting correct metadata from the media package. Always extract it from the Dublin Core catalog. My first test ended up like this:

![Screenshot from 2022-04-20 16-40-36](https://user-images.githubusercontent.com/1008395/164257290-55b03ff8-468b-448f-b319-71a9be897b18.png)

Reply:
Maybe something like:

```diff
@@ -255,15 +255,17 @@ public class Ingestor implements Runnable {
   try {
     Recording recordingState = schedulerService.getRecordingState(id);
     if (recordingState.getState().equals(UPLOAD_FINISHED)) {
+      var referenceId = mediaPackage.getIdentifier().toString();
       mediaPackage = (MediaPackage) mediaPackage.clone();
-      Id newId = IdImpl.fromUUID();
-      mediaPackage.setIdentifier(newId);
+      mediaPackage.setIdentifier(IdImpl.fromUUID());

       // Update dublincore title
       try {
-        String newTitle = mediaPackage.getTitle() + " - " + newId;
         DublinCoreCatalog dc = DublinCoreUtil.loadEpisodeDublinCore(workspace, mediaPackage).get();
+        var newTitle = dc.get(DublinCore.PROPERTY_TITLE).get(0).getValue()
+            + " (" + Instant.now().getEpochSecond() + ")";
         dc.set(DublinCore.PROPERTY_TITLE, newTitle);
+        dc.set(DublinCore.PROPERTY_REFERENCES, referenceId);
         mediaPackage = updateDublincCoreCatalog(mediaPackage, dc);
         mediaPackage.setTitle(newTitle);
       } catch (Exception e) {
```

This…
- uses the DC title
- adds the original event as reference
- switches from the very long UUID to the ingest time (as unix timestamp) which will at least give you an order for the ingested events

Reply:
Very good suggestions, should be implemented now.